src/graphics/text.c

```diff
@@ -84,6 +86,40 @@ int text_get_width(const uint8_t *str, font_t font)
     return width;
 }
 
+void text_fit_to_width(char *str, font_t font, int requested_width, const char *end_replacement)
+{
+    char *orig_str = str;
+    const font_definition *def = font_definition_for(font);
+    int end_replacement_width = (end_replacement) ? text_get_width((const uint8_t *)end_replacement, font) : 0;
+    unsigned int maxlen = 10000;
+    int width = 0;
+    int length_with_replacement = 0;
+    int image_base = image_group(GROUP_FONT);
+    while (*str && maxlen > 0) {
+        if (*str == ' ') {
+            width += def->space_width;
+        } else {
+            int image_offset = font_image_for(*str);
+            if (image_offset) {
+                int image_id = image_base + def->image_offset + image_offset - 1;
+                width += def->letter_spacing + image_get(image_id)->width;
+            }
+        }
+        if (end_replacement_width + width <= requested_width) {
+            length_with_replacement++;
+        }
+        if (width > requested_width) {
+            break;
+        }
+        str++;
+        maxlen--;
+    }
+    if (10000 - maxlen < strlen(orig_str))
+    {
```

Comment:
Curly brace on the same line as the `if`, please :)

Reply:
Old habits die hard!!! Your coding style is more densely packed than mine and it takes actual effort to keep things consistant.